src/app/contact-list/edit-contact/edit-contact.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Contact } from 'src/app/contacts.service';
+import { ContactsService } from 'src/app/contacts.service';
+
+@Component({
+  selector: 'app-edit-contact',
+  templateUrl: './edit-contact.component.html',
+  styleUrls: ['./edit-contact.component.css'],
+})
+export class EditContactComponent implements OnInit {
+  contactsService: ContactsService;
+  id: number = -1;
+  user: Contact | undefined;
+  contactForm: FormGroup;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    contactsService: ContactsService
+  ) {
+    this.contactsService = contactsService;
+    this.contactForm = new FormBuilder().group({
+      firstname: ['', Validators.required],
+      lastname: ['', Validators.required],
+      street: ['', Validators.required],
+      city: ['', Validators.required],
+    });
+  }
+
+  ngOnInit() {
+    this.route.params.subscribe((params) => {
+      this.id = +params['id'];
+    });
+    this.user = this.contactsService.getContact(this.id);
+    if (this.user) {
+      this.contactForm = new FormBuilder().group({
+        firstname: [this.user?.firstname, Validators.required],
+        lastname: [this.user?.lastname, Validators.required],
+        street: [this.user?.street, Validators.required],
+        city: [this.user?.city, Validators.required],
+      });
+    } else {
+      console.error('User not found');
+      this.router.navigate(['/contacts']);
+    }
+  }
+
+  onSubmit() {
+    console.warn('User Edited', this.contactForm);
+    this.contactsService.editContact(this.id, this.contactForm.value);
```

Comment:
After successfully editing the user, you might want to navigate back to a contacts list page or the detail page of the edited contact

Reply:
Prevent the form from being resubmitted if it's already processing a submission. This can be achieved by disabling the submit button or by keeping a boolean flag in your component state